File: imglyKit/IMGLYFoodFilter.m
//
//  IMGLYFoodFilter.m
//  imglyKit
//
//  Created by Carsten Przyluczky on 22.08.13.
//  Copyright (c) 2013 9elements GmbH. All rights reserved.
//

#import "IMGLYFoodFilter.h"

@implementation IMGLYFoodFilter

- (id)init {
    self = [super init];
    if (self) {
        GPUImageSaturationFilter *saturationFilter = [[GPUImageSaturationFilter alloc] init];
        [saturationFilter setSaturation:1.35];
        
        GPUImageContrastFilter *contrastFilter = [[GPUImageContrastFilter alloc] init];
        [contrastFilter setContrast:1.1];
        
        [self addFilter:saturationFilter];
        [self addFilter:contrastFilter];
        
        [saturationFilter addTarget:contrastFilter];
        
        [self setInitialFilters:@[saturationFilter]];
        [self setTerminalFilter:contrastFilter];
    }
    
    return self;
}

@end

File: Pod/Classes/IMGLYFoodFilter.swift
//
//  IMGLYFoodFilter.swift
//  imglyKit
//
//  Created by Carsten Przyluczky on 11/03/15.
//  Copyright (c) 2015 9elements GmbH. All rights reserved.
//

import Foundation

public class IMGLYFoodFilter: IMGLYResponseFilter {
    override public init() {
        super.init()
        self.responseName = "Food"
        self.displayName = "food"
    }
    
    required public init(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
    }
    
    override public var filterType:IMGLYFilterType {
        get {
            return IMGLYFilterType.Food
        }
    }
}
